lib/project_types/script/layers/infrastructure/script_project_repository.rb

```diff
@@ -10,6 +10,21 @@ class ScriptProjectRepository
         SCRIPT_JSON_FILENAME = "script.json"
         MUTABLE_ENV_VALUES = %i(uuid)
 
+        def self.create_project_directory(ctx:, directory:)
```

Comment:
Any reason you went with class methods instead of instance methods?

Reply:
True, it might be better to have these instance methods and initializer like:
```
init(ctx: init_dir:, dir:)
```

Reply:
Oh another funny thing: I don't see changes to `FakeScriptProjectRepo`?  POODR warned about this meta-verse problem?  🤔

Reply:
I'm not sure why I went with class vs instance. I've changed it to use instance methods, and also just use the Repo's own context instead of needing to pass it along everywhere.